### PR TITLE
Feature/support hiding search button

### PR DESCRIPTION
**Pull request by minevala**
## Description
- Add support to hide search button

## Related Issues
- https://helsinkisolutionoffice.atlassian.net/browse/HDS-725

## Motivation and Context
- This is part of the larger concept of adding custom search button functionality. In some cases, using a custom search button might be more obvious for the user

## How Has This Been Tested?
- locally in dev machine
- in loki tests

👉 [Storybook Demo](https://city-of-helsinki.github.io/hds-demo/search-input-hide-button/?path=/story/components-searchinput--without-custom-search-button)

## Screenshots (if appropriate):
Screenshot 2021-12-02 at 16 24 44

### CODE REVIEW

**Review by laurakarhu — APPROVED** [review]
Nice!

**Comment by minevala** [comment]
I think we need to take this into account as well: https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers

File: packages/react/src/components/searchInput/SearchInput.tsx [hunk 1]
```diff
@@ -148,24 +154,30 @@ export const SearchInput = <SuggestionItem,>({
...
-          )}
-          <button
-            type="button"
-            aria-label={searchButtonAriaLabel}
-            className={classNames(styles.button)}
-            onClick={submit}
-          >
-            <IconSearch className={styles.searchIcon} aria-hidden />
-          </button>
+          <>
```
Comment (VilleMiekkoja) [inline packages/react/src/components/searchInput/SearchInput.tsx:157]: This Fragment is probably not needed?

**Review by VilleMiekkoja — COMMENTED** [review]
Good job, one minor comment. 👍

File: packages/react/src/components/searchInput/SearchInput.tsx (hunk 1 above)
Comment (minevala) [reply packages/react/src/components/searchInput/SearchInput.tsx]: - [x] Remove unnecessary React fragments

**Review by ronijaakkola — APPROVED** [review]
Works well!

**Pull request merged**